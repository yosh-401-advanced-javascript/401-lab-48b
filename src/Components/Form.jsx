import React from 'react';

class Form extends React.Component {
  render() {
    return (
        <div>
          <form onSubmit={this.props.counter}>
            <input
                type="text"
                value={this.props.name}
                onChange={this.props.handleChange}
            />
          </form>
          <button type="submit"> Click Me! </button>
        </div>
    );
  }
}


export default Form;
