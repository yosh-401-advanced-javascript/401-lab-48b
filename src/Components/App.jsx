import React from 'react';
import Form from './Form';

class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      count: 0,
      name: '',
    };
  }

  counter = (event) => {
    console.log('here');
    this.setState((previousState) => {
      return {count: previousState.count + 1}
    });

  };

  handleChange = event => {
    console.log(event.target.value);
    this.setState( {
      name: event.target.value,

    });
  };

  render() {
    return (
        <React.Fragment>
          <div>
            <Form
                counter={this.counter}
                handleChange={this.handleChange}
                count={this.state.count}
                name={this.state.name}
            />
            <div>Name:  {this.state.name} </div>
            <div className='name'># of Updates:  {this.state.count} </div>
          </div>
        </React.Fragment>
    );
  }
}


export default App;
