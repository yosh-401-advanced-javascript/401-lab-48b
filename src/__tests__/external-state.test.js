import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from '../Components/App.jsx';
import Form from '../Components/Form.jsx';

Enzyme.configure({ adapter: new Adapter() });
global.shallow = shallow;
global.mount = mount;


describe('Testing Counter', () => {
  it('is alive at application start', () => {
    const app = shallow(<App />);
    expect(app.find('div').exists()).toBeTruthy();
  });
});

it('should render correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should increment the count', () => {
  const form = shallow(<Form />);
  const app = shallow(<App />);
  const incrementButton = form.find('button');
  incrementButton.simulate('click');
  const text = app.find('.name').text();
  expect(text).toEqual('# of Updates:  0 ');
});

describe('Testing App ', () => {
  it('should should render correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  describe('Testing Form ', () => {
    it('should should render correctly', () => {
      const tree = renderer.create(<Form />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
