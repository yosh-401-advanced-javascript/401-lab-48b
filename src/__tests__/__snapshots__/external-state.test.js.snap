// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App  Testing Form  should should render correctly 1`] = `
<div>
  <form>
    <input
      type="text"
    />
  </form>
  <button
    type="submit"
  >
     Click Me! 
  </button>
</div>
`;

exports[`Testing App  should should render correctly 1`] = `
<div>
  <div>
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </form>
    <button
      type="submit"
    >
       Click Me! 
    </button>
  </div>
  <div>
    Name:  
    
     
  </div>
  <div
    className="name"
  >
    # of Updates:  
    0
     
  </div>
</div>
`;

exports[`should render correctly 1`] = `
<div>
  <div>
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </form>
    <button
      type="submit"
    >
       Click Me! 
    </button>
  </div>
  <div>
    Name:  
    
     
  </div>
  <div
    className="name"
  >
    # of Updates:  
    0
     
  </div>
</div>
`;
